.changelog/unreleased/breaking-changes/4642-close-txindexer.md

```diff
@@ -0,0 +1,2 @@
+- `[state]` `TxIndexer` no longer leaks its database's goroutines.
```

Comment:
The change also adds a `Close` method to the `TxIndexer` interface, though this is the main benefit of this PR. Can I just add that point to the same changelog here?

Reply:
This should be state here in the message.